src/FileStorage/BucketFileStorage.php

```diff
@@ -0,0 +1,175 @@
+<?php
+
+/**
+ * Copyright © 2016-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+declare(strict_types=1);
+
+namespace SprykerSdk\Integrator\FileStorage;
+
+use Aws\Credentials\Credentials;
+use Aws\S3\S3Client;
+use RuntimeException;
+use SprykerSdk\Integrator\IntegratorConfig;
+
+class BucketFileStorage implements BucketFileStorageInterface
+{
+    /**
+     * @var string
+     */
+    protected const BUCKET_KEY = 'Bucket';
+
+    /**
+     * @var string
+     */
+    protected const OBJECT_KEY = 'Key';
+
+    /**
+     * @var string
+     */
+    protected const BODY_KEY = 'Body';
+
+    /**
+     * @var string
+     */
+    protected const VERSION_KEY = 'version';
+
+    /**
+     * @var string
+     */
+    protected const LATEST_KEY = 'latest';
+
+    /**
+     * @var string
+     */
+    protected const REGION_KEY = 'region';
+
+    /**
+     * @var string
+     */
+    protected const CREDENTIALS_KEY = 'credentials';
+
+    /**
+     * @var string
+     */
+    protected const CONTENT_LENGTH_KEY = 'ContentLength';
+
+    /**
+     * @var \Aws\S3\S3Client $s3Client
+     */
+    protected ?S3Client $s3Client = null;
+
+    /**
+     * @var \SprykerSdk\Integrator\IntegratorConfig $config
+     */
+    protected IntegratorConfig $config;
+
+    /**
+     * @param \SprykerSdk\Integrator\IntegratorConfig $config
+     */
+    public function __construct(IntegratorConfig $config)
+    {
+        $this->config = $config;
+    }
+
+    /**
+     * @param string $filePath
+     *
+     * @return string|null
+     */
+    public function getFile(string $filePath): ?string
+    {
+        $client = $this->getS3Client();
```

Comment:
can be moved down, after the bucket name check

Reply:
in this case, the public method will be under the protected one - the situation is restricted by your clean-code rules